core/discord/src/main/java/app/otakureader/core/discord/DiscordRpcService.kt

```
@@ -0,0 +1,245 @@
+
+    /**
+     * Disconnect from Discord.
+     */
+    fun disconnect() {
+        scope.launch {
+            try {
+                kizzyRpc?.closeRPC()
+                kizzyRpc = null
+                _connectionState.value = ConnectionState.Disconnected
+                currentActivity = null
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)